app/src/main/java/uk/co/neuralcubes/neuralates/controller/RobotController.java

```diff
@@ -15,33 +15,51 @@
     private static final String TAG = "ROBOT_CONTROLLER";
     final private ConvenienceRobot mRobot;
     final private EventBus mBus;
+    private ColorMap mColorMap;
+
 
     private static final double MAX_THRUST = 0.1; //valid values from 0 to 1
     private double mConcentration = 0.0;
     private boolean mOverrideFocus = false;
 
-    public RobotController(@NonNull ConvenienceRobot robot ,@NonNull EventBus bus) {
+    public RobotController(@NonNull ConvenienceRobot robot,@NonNull EventBus bus,
+                           @NonNull ColorMap colorMap) {
         mRobot = robot;
         mBus = bus;
+        mColorMap = colorMap;
         mBus.register(this);
+
     }
 
     public void unlink(){
         mBus.unregister(this);
     }
 
     @Subscribe
-    public synchronized void updateAcelerometer(MuseHandler.AccelerometerReading reading) {
+    public synchronized void updateAccelerometer(MuseHandler.AccelerometerReading reading){
+
         mRobot.drive(computeAngle(reading.getX(),reading.getY()),getThrust());
+
     }
 
     float getThrust(){
             return (float)( mConcentration * MAX_THRUST);
     }
 
+
+
     @Subscribe
     public synchronized void updateConcentration(MuseHandler.FocusReading reading) {
         mConcentration = mOverrideFocus?(MAX_THRUST*0.7):reading.getFocus();
+        int []color = this.mColorMap.map(mConcentration);
+        mRobot.setLed(color[0]/255.f,color[1]/255.f,color[2]/255.f);
+    }
+    /**
+     * this might come from the ui while the rest of the updates from the muse
+     */
+
+    public synchronized void setColorMap(ColorMap colorMap) {
+        mColorMap = colorMap;
```

Comment:
Reference assignments are atomic in Java so there's no need for this to be synchronized.